apps/calendar/js/app.js

```diff
@@ -23,225 +24,185 @@ var storeFactory = require('store/factory');
-   *  - endEvent ( ditto )
-   *  - pending
-   *
-   * @param {Object} object to observe.
-   */
-  observePendingObject: function(object) {
-    this._pendingManager.register(object);
-  },
-
-  isPending: function() {
-    return this._pendingManager.isPending();
-  },
-
-  /**
-   * Internally restarts the application.
-   */
-  forceRestart: function() {
-    debug('Will restart calendar app.');
-    window.location.href = this.startingURL;
-  },
-
-  _routes: function() {
-
-    /* routes */
-    router.state('/week/', 'Week');
-    router.state('/day/', 'Day');
-    router.state('/month/', ['Month', 'MonthDayAgenda']);
-    router.modifier('/settings/', 'Settings', { clear: false });
-    router.modifier('/advanced-settings/', 'AdvancedSettings', {
-      color: 'settings'
+function renderView(viewName) {
+  return new Promise(accept => {
+    viewFactory.get(viewName, view => {
+      view.render();
+      accept();
     });
+  });
+}
 
-    router.state('/alarm-display/:id', 'ViewEvent', { path: false });
-
-    router.state('/event/add/', 'ModifyEvent');
-    router.state('/event/edit/:id', 'ModifyEvent');
-    router.state('/event/show/:id', 'ViewEvent');
-
-    router.modifier('/select-preset/', 'CreateAccount');
-    router.modifier('/create-account/:preset', 'ModifyAccount');
-    router.modifier('/update-account/:id', 'ModifyAccount');
-
-    router.start();
-
-    // at this point the tabs should be interactive and the router ready to
-    // handle the path changes (meaning the user can start interacting with
-    // the app)
-    performance.chromeInteractive();
-
-    var pathname = window.location.pathname;
-    // default view
-    if (pathname === '/index.html' || pathname === '/') {
-      router.go('/month/');
+/**
+ * Primary code for app can go here.
+ */
+function init() {
+  debug('Will initialize calendar app');
+  setupCore();
+
+  pendingManager.onpending = () => {
+    document.body.classList.add('pending-operation');
+  };
+
+  pendingManager.oncomplete = () => {
+    document.body.classList.remove('pending-operation');
+    performance.pendingReady();
+    if (!loadLazyStyles) {
+      // XXX: not loading the 'lazy_loaded.js' here anymore because for some
+      // weird reason curl.js was returning an object instead of
+      // a constructor when loading the "views/view_event" when starting the
+      // app from a notification; might be related to the fact we bundled
+      // multiple modules into the same file, are using the "paths" config to
+      // set the location and also using the async require in 2 places and
+      // using different module ids for each call.. the important thing is
+      // that this should still give a good performance result and works as
+      // expected.
+      loadLazyStyles = asyncRequire('css!lazy_loaded');
     }
+  };
 
-  },
 
-  _initControllers: function() {
-    // controllers can only be initialized after db.load
+  pendingManager.register(core.syncController);
```

Comment:
nit: use a single empty line before and after this (currently there are 2 empty lines before)